aiohttp/cookiejar.py

```diff
@@ -110,20 +112,21 @@ def clear(self, predicate: Optional[ClearCookiePredicate] = None) -> None:
 
         to_del = []
         now = datetime.datetime.now(datetime.timezone.utc)
-        for domain, cookie in self._cookies.items():
+        for (domain, path), cookie in self._cookies.items():
             for name, morsel in cookie.items():
-                key = (domain, name)
+                key = (domain, path, name)
                 if (
                     key in self._expirations and self._expirations[key] <= now
                 ) or predicate(morsel):
                     to_del.append(key)
 
-        for domain, name in to_del:
+        for domain, path, name in to_del:
             key = (domain, name)
             self._host_only_cookies.discard(key)
```

Comment:
```suggestion
            self._host_only_cookies.discard((domain, name))
```

As this is not being reused...

Reply:
Should also resolve your mypy error.

Reply:
Should also resolve your mypy error.